scripts/scil_project_streamlines_to_map.py

```diff
@@ -0,0 +1,232 @@
+    for idx, s in enumerate(streamlines):
+        mins.append(np.min(s.astype(int), 0))
+        maxs.append(np.max(s.astype(int), 0) + 1)
+        if just_endpoints:
+            s = np.stack((s[0, :], s[-1, :]), axis=0)
+        offset_streamlines.append((s - mins[-1]).astype(np.float32))
+
+    offset_streamlines = ArraySequence(offset_streamlines)
+    indices = uncompress(offset_streamlines)
+
+    return mins, maxs, indices
+
+
+def _project_metrics(curr_metric_map, count, orig_s, streamline_mean,
+                     just_endpoints):
+    if just_endpoints:
+        xyz = orig_s[0, :].astype(int)
+        curr_metric_map[xyz[0], xyz[1], xyz[2]] += streamline_mean
+        count[xyz[0], xyz[1], xyz[2]] += 1
+
+        xyz = orig_s[-1, :].astype(int)
+        curr_metric_map[xyz[0], xyz[1], xyz[2]] += streamline_mean
+        count[xyz[0], xyz[1], xyz[2]] += 1
+    else:
+        for x, y, z in orig_s[:].astype(int):
+            curr_metric_map[x, y, z] += streamline_mean
+            count[x, y, z] += 1
+
+
+def _pick_data(args, sft):
+    if args.use_dps or args.load_dps:
+        if args.use_dps:
+            for dps in args.use_dps:
+                if dps not in sft.data_per_streamline:
+                    raise IOError('DPS key not in the sft: {}'.format(dps))
+            name = args.use_dps
+            data = [sft.data_per_streamline[dps] for dps in args.use_dps]
+        else:
+            name = args.load_dps
+            data = [load_matrix_in_any_format(dps) for dps in args.load_dps]
+        for i in range(len(data)):
+            if len(data[i]) != len(sft):
+                raise IOError('DPS length does not match the SFT: {}'
+                              .format(name[i]))
+    elif args.use_dpp or args.load_dpp:
+        if args.use_dpp:
+            name = args.use_dpp
+            data = [sft.data_per_point[dpp]._data for dpp in args.use_dpp]
+        else:
+            name = args.load_dpp
+            data = [load_matrix_in_any_format(dpp) for dpp in args.load_dpp]
+        for i in range(len(data)):
+            if len(data[i]) != len(sft.streamlines._data):
+                raise IOError('DPP length does not match the SFT: {}'
+                              .format(name[i]))
+    return zip(name, data)
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, [args.in_bundle], args.in_metrics +
+                        args.load_dps + args.load_dpp)
+    assert_output_dirs_exist_and_empty(parser, args,
+                                       args.out_folder,
+                                       create_dir=True)
+
+    sft = load_tractogram_with_reference(parser, args, args.in_bundle)
+    sft.to_vox()
+    sft.to_corner()
+
+    if len(sft.streamlines) == 0:
+        logging.warning('Empty bundle file {}. Skipping'.format(args.bundle))
+        return
+
+    mins, maxs, indices = _process_streamlines(sft.streamlines,
```

Comment:
This and the following lines should be abstracted in a or multiple functions, which can reside in `scilpy`. The function(s) could return maps that can be saved in the main function. Overall, only IO should be in the main function.